tests/testsuite/git.rs

```diff
@@ -3673,14 +3673,14 @@ fn different_user_relative_submodules() {
             "\
 [UPDATING] git repository `{}`
 [UPDATING] git submodule `{}`
-[UPDATING] git submodule `{}`
+[UPDATING] git submodule `{}/../dep2`
```

Comment:
The regression of URL display is one downside of it. However, since we drop `Url::parse` here so now we can have relative submodule inside a scp-like submodule. For example, this was not possible before this PR:

```
direct git dependency
  submodule url `git@github.com/user/repo`
     submodule url `../`
```